api/v1beta1/cluster_types.go

```diff
@@ -686,7 +686,7 @@ type MachineDeploymentTopology struct {
 	// the values are hashed together.
 	// +required
 	// +kubebuilder:validation:MinLength=1
-	// +kubebuilder:validation:MaxLength=256
+	// +kubebuilder:validation:MaxLength=64
```

Comment:
PR description says 63, why 64 here?

Reply:
I thought that 256 here meant "strictly less than 256." Looks like I was incorrect, although actually confirming that was not straightforward.

The kubebuilder MaxLength annotation becomes the `maxLength` OpenAPI field. According to [this](https://github.com/OAI/OpenAPI-Specification/issues/1754), `maxLength` is [specified](https://datatracker.ietf.org/doc/html/draft-wright-json-schema-validation-00#section-5.6) by JSON Schema:
> A string instance is valid against this keyword if its length is less than, or equal to, the value of this keyword.

I'll change it to 63.

Reply:
(As an aside, I'm not sure where the 256 limit comes from. Maybe it was meant to be 253? That's the highest limit I see in https://kubernetes.io/docs/concepts/overview/working-with-objects/names/)

Reply:
Fixed in https://github.com/kubernetes-sigs/cluster-api/pull/12072/commits/3438ebf42b2660bf2ae7ddf5d4da5c5dca25927b

Reply:
256 was just a value that seemed "high enough" for whatever folks used before.

It intentionally did not correlate to any max CRD name length.

(as mentioned on the other PR, I was not aware of the pre-existing validation, so I assumed folks could have used pretty long values here as the MD name generation cuts the generated name off)